### ISSUE TITLE
Work on stateless components without .jsx extension

### ISSUE THREAD

**Issue by clayne11** [gadicc/meteor-react-hotloader#18]
I use `.js` extension everywhere  throughout my application. Is it possible to add hot-reloading to stateless functional components that just use the `.js` extension but follow the rest of the rules, ie imports React and returns JSX syntax?

**Comment by gadicc** [comment]
This will be in the next release.  I'm a bit nervous about false positives but let's see how we go.  I think this was inevitable, a lot of people no longer rely on the `.jsx` extensions.

**Comment by gadicc** [comment]
Just a quick update, this was planned for a future release but I had to release a critical fix and didn't want this to be mixed with it, so I reverted it https://github.com/gadicc/meteor-react-hotloader/commit/48d5edbf9bab69c536be2eae99eec43b3624f731.  This is still pegged for the next release, but I'd like to spend a bit more time making it optional rather than forcing it.  Will keep you posted.

**Comment by mordrax** [comment]
+1 on this, I didn't realise `.jsx` was a requirement and have been moving all my components to `.js`

**Comment by gadicc** [comment]
Yeah, as soon as I have time to finish it off this will be via options in package.json.

**Comment by gadicc** [comment]
From the latest [experimental release](https://github.com/gadicc/meteor-react-hotloader/issues/26#issuecomment-207743140), you can now configure this via the `package.json`:

```js
  "ecmascript-hot": {
    "transformStateless": {
      "pathMatch": "\\.jsx?$",
    }
  }
```

It will be in the next stable release.  More details in the [README](https://github.com/gadicc/meteor-react-hotloader/tree/faster#readme) from the [faster](https://github.com/gadicc/meteor-react-hotloader/tree/faster) branch (you can also set a `sourceMatch` pattern to whitelist/blacklist based on the contents of the file).

**Comment by gadicc** [comment]
Oops, will keep open until it's published in a stable release.

**Comment by sammkj** [comment]
@gadicc I couldn't get stateless component to be hot reloaded. I'm using fast.5. I also have the config in my `package.json`.

**Comment by gadicc** [comment]
Ugh :(  There's a caching issue here.  Will get it fixed in next release.  For now, if you change the `ecmascript-hot` settings in `package.json`, you need to also:

```sh
$ rm .meteor/local/plugin-cache/gadicc_ecmascript-hot/local/*.json
```

in your project dir before starting Meteor again.  (oh and yes, restart Meteor... I'll make the new ver auto quit too).

**Comment by gadicc** [comment]
Fix published in `gadicc:ecmascript-hot@1.3.1-fast.6` (https://github.com/gadicc/meteor-react-hotloader/commit/3cbf2eb3a1a1245226beb24bf0a4c3f3ba68e47a)
